exec/java-exec/src/main/java/org/apache/drill/exec/server/Drillbit.java

```diff
@@ -218,13 +218,18 @@ private void javaPropertiesToSystemOptions() {
   }
 
   /**
-   * If cluster version is the same as default, update option to current drillbit version.
+   * If cluster version is the same as default, updates option to current drillbit version.
+   * Since getOption and setOption are separate calls, raise condition might occur
+   * when several drillbits are registering at the same time.
+   * It is assumed that the first drillbit in cluster sets cluster version
+   * but when the raise condition occurs, it might be the second or the third etc.
+   * This behaviour does not impose significant impact, since the main goal is to detect mismatching versions.
    */
-  private void checkAndUpdateClusterVersionOption(OptionManager optionManager, DrillbitEndpoint drillbitEndpoint) {
-    OptionValue versionOption = optionManager.getOption(ExecConstants.CLUSTER_VERSION);
+  private void checkAndUpdateClusterVersionOption(OptionManager systemOptions, DrillbitEndpoint drillbitEndpoint) {
```

Comment:
How can this work? Consider this:

1. Start a new cluster.
2. A Drillbit registers. It sets the cluster version.
3. A second Drillbit registers. Does it reset the version? Or, does it accept that the first has already set the version?
4. Shut down the cluster.
5. Upgrade Drill to a new version.
6. Start the first Drillbit.
7. Does this Drillbit overwrite the cluster version? Or, accept the existing (old) version?

Note that steps 3 and 7 are identical from ZK's perspective.